node/network/collator-protocol/src/error.rs

```diff
@@ -0,0 +1,94 @@
+// Copyright 2021 Parity Technologies (UK) Ltd.
+// This file is part of Polkadot.
+
+// Polkadot is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Polkadot is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Polkadot.  If not, see <http://www.gnu.org/licenses/>.
+//
+
+//! Error handling related code and Error/Result definitions.
+
+use polkadot_node_primitives::UncheckedSignedFullStatement;
+use polkadot_subsystem::SubsystemError;
+use thiserror::Error;
+
+use polkadot_node_subsystem_util::{Fault, runtime, unwrap_non_fatal};
+
+use crate::LOG_TARGET;
+
+/// General result.
+pub type Result<T> = std::result::Result<T, Error>;
+/// Result for non fatal only failures.
+#[allow(dead_code)]
+pub type NonFatalResult<T> = std::result::Result<T, NonFatal>;
```

Comment:
Where is this used?

Reply:
It is not, but I still want to have it as a general framework code. E.g. every subsystem has `Fatal` and `NonFatal` errors and a `NonFatalResult`, .... In the spirit of consistency I would like to leave it, although it is not used right now. I would definitely like to avoid for example, if it is needed at one point that people make up a new name.

Reply:
Then you should write something in the guide, but dead code isn't a good practice and if someone needs this, it is restored in seconds.